### PR TITLE
adding the changes related to module2

### PR DESCRIPTION
**Pull request by HamidurRahman1**
@sargent-bu This PR covers the work (changes) related to the following pivotal stories - 

- https://www.pivotaltracker.com/n/projects/2626708/stories/184310504
- https://www.pivotaltracker.com/n/projects/2626708/stories/184310541
- https://www.pivotaltracker.com/n/projects/2626708/stories/184310516
- https://www.pivotaltracker.com/n/projects/2626708/stories/184310563
- https://www.pivotaltracker.com/n/projects/2626708/stories/184311082
- https://www.pivotaltracker.com/n/projects/2626708/stories/184331260

### CODE REVIEW

**Review by nbnkiran — COMMENTED** [review]
Please add few comments in the code so it easy to understand the purpose or the feature of the code.

File: src/main/java/edu/bu/cs633/minimalsearchengine/utils/Utilities.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import edu.bu.cs633.minimalsearchengine.exceptions.customExceptions.ConstraintViolationException;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public final class Utilities {
+
+    public static final Set<String> STOP_WORDS = new HashSet<>(Arrays.asList("i", "me", "my", "myself", "we",
```
Comment (s-gorrin) [inline src/main/java/edu/bu/cs633/minimalsearchengine/utils/Utilities.java:11]: Are stop words checked before or after input sanitation? I assume after, in which case the apostrophe (') character should never make it this far, so the contractions here ("it's", "you've") should be modified to however we end up handling sanitizing input. If, for example, an input of "you've" becomes "youve" then that should be the stop word. If "you've" instead becomes "you" and "ve" then that is a different handling.

**Review by s-gorrin — COMMENTED** [review]
I agree that at least a comment per file would be helpful. I tend to over-comment my code due to spending so much time in academia, but it's nice to know what methods do when they're not obvious to a casual observer. Otherwise, no objections to the merge.

File: src/main/java/edu/bu/cs633/minimalsearchengine/utils/Utilities.java (hunk 1 above)
Comment (HamidurRahman1) [reply src/main/java/edu/bu/cs633/minimalsearchengine/utils/Utilities.java:11]: @s-gorrin  I thought about the STOP_WORDS handling as well. I have not decided to how to use/handle the query from user containing (you've). That's what I was planning on bringing up to the others this Tuesday, but at least for sure we will use this set of words as stop words, and that's why I added them as part of this PR.

**Comment by HamidurRahman1** [comment]
@sargent-bu @nbnkiran @s-gorrin I also agree about commenting but haven't focused on that much yet but should have. We can surely discuss this Tuesday along with others

**Comment by sargent-bu** [comment]
I completely agree with @s-gorrin 's comments re: in-line source comments for developer documentation. If we had to pass this off to another dev or reshuffle responsibilities, the next devs may be at a disadvantage despite how straightforward the code is thus far. 

@HamidurRahman1 
I'm going to throw a few stories under the appropriate epic to reflect this. 

From what I can see the database design is preliminary. I would like to talk about this in the upcoming meeting on Tuesday. I think we can build upon this.

**Pull request merged**